xlators/cluster/afr/src/afr-open.c

```diff
@@ -311,9 +325,50 @@ afr_fix_open(fd_t *fd, xlator_t *this)
     ret = loc_path(&local->loc, NULL);
     if (ret < 0)
         goto out;
+    local->cont.lk.getlk_rsp1 = GF_CALLOC(sizeof(*local->cont.lk.getlk_rsp1),
+                                          priv->child_count, gf_afr_mt_gf_lock);
 
     local->fd = fd_ref(fd);
     local->fd_ctx = fd_ctx;
+    local->strict_locks = _gf_false;
+    local->op = GF_FOP_OPEN;
+    flock.l_type = F_WRLCK;
+    afr_set_lk_owner(frame, this, frame->root);
+    flock.l_owner.len = frame->root->lk_owner.len;
+    if (flock.l_owner.len > 0) {
+        memcpy(flock.l_owner.data, frame->root->lk_owner.data,
+               flock.l_owner.len);
+    }
+
+    AFR_ONLIST(priv->child_up, frame, afr_getlk_cbk, lk, fd, F_GETLK, &flock,
+               xdata);
+
+    success_replies = alloca0(priv->child_count);
+    afr_fill_success_replies(local, priv, success_replies);
+    if (AFR_COUNT(success_replies, priv->child_count) == 0) {
+        ret = -1;
+        goto out;
+    }
+
+    for (i = 0; i < priv->child_count; i++) {
+        if (!local->replies[i].valid || local->replies[i].op_ret != 0)
+            continue;
+        if (local->cont.lk.getlk_rsp1[i].l_type == F_UNLCK) {
+            continue;
+        } else if (local->cont.lk.getlk_rsp1[i].l_type == F_WRLCK &&
+                   !is_same_lkowner(&local->cont.lk.getlk_rsp1[i].l_owner,
+                                    flock.l_owner.data)) {
+            locks_held = _gf_true;
+            break;
+        }
+    }
+
+    if (local->strict_locks && locks_held) {
+        gf_msg_debug(this->name, 0,
```

Comment:
I think INFO level is better. We need to print the gfid too.

Reply:
Done.